doc: add doc for osm-to-railjson (#344)
DGEXSolutions/osrd-website: content/docs/guides/osm_infra/open-data.en.md

---

> title: "Working with open data"

Reviewer: ```suggestion
title: "Working with OSM data"
```

Author: 6a016b4612d0c2dcad8205d6d547e91a1a4f5606

> linkTitle: "Working with open data"

Reviewer: ```suggestion
linkTitle: "Working with OSM data"
```

Author: 6a016b4612d0c2dcad8205d6d547e91a1a4f5606

weight: 10

> description: Describe the problems and solutions related to working with open data using OSRD

Reviewer: ```suggestion
description: Describe the problems and solutions related to working with OSM data using OSRD
```

Author: 6a016b4612d0c2dcad8205d6d547e91a1a4f5606

> description: Describe the problems and solutions related to working with open-data inside OSRD

Reviewer 2: ```suggestion
description: Describe the problems and solutions related to working with open data using OSRD
```

---

### Problems

> Some data are absent from OSM for a few reasons:

Reviewer 2: ```suggestion
Some data are not available from OpenStreetMap (OSM) for different reasons:
```

> Some data are not available from OpenStreetMap (OSM) for different reasons:

Reviewer: ```suggestion
Some data is not available from OpenStreetMap (OSM) for different reasons:
```

Author: 6a016b4612d0c2dcad8205d6d547e91a1a4f5606

> - It can be private, like *signals* in France.

Reviewer 2: ```suggestion
- data can be private, like signals in France,
```

> - It can be missing, like trigram in France, which are public but no one has taken the time to recorde them properly in OSM.

Reviewer 2: ```suggestion
- data can be missing, like trigrams in France, which are public but no one has taken the time to record them properly in OSM,
```

> - data can be missing, like trigrams in France, which are public but no one has taken the time to record them properly in OSM,

Reviewer: are trigrams really public?

Author: I assumed, but I don't know for sure

Reviewer: I don't think so

> - It can also be badly recorded, like in Morocco where a lot of track have wrong associated tags.

Reviewer 2: ```suggestion
- data can also be badly recorded, like in Morocco where a lot of tracks have the wrong associated tags,
```

> - data can also be badly recorded, like in Morocco where a lot of tracks have the wrong associated tags,

Reviewer: why is that? do you have some examples? this could help the OSM community correct it.

Author: The main issue I had was with tracks with the wrong `service` tag. They had `service=yard` or `service=siding` instead of `service=crossover`, which resulted in us wrongly filtering them. I corrected those I found (like 4 of them I think).

Reviewer: I would replace all these bullet points with a more general sentence then, something like "Some data is not available from OpenStreetMap (OSM), either because it's not public or because it has not been collected in OSM yet."

Author: Done, I kept the example because Louis seemed attached to it.
cf895200d38e6de815ad4d7c88b6f40ffa19a78a

> - And lastly, OSM doesn't support every data. For example, *slopes* are not stored in OSM.

Reviewer 2: ```suggestion
- and lastly, OSM doesn't support every object, for example, slopes are not stored in OSM.
```

### OSM data in OSRD

> Here is the list of the data present in OSRD, roughly sorted by there importance in the application.</br>

Reviewer 2: ```suggestion
The following list resumes the different objects used in OSRD, roughly sorted by their importance in the application.</br>
```

> It gives a rough idea about their role in OSRD and how they are extracted (or not) from OSM.

Reviewer 2: ```suggestion
This gives a rough idea about their role in OSRD and how they are extracted (or not) from OSM.
```

> This gives a rough idea about their role in OSRD and how they are extracted (or not) from OSM.

Reviewer: ```suggestion
This gives an idea about their role in OSRD and how they are extracted (or not) from OSM.
```

Author: 6a016b4612d0c2dcad8205d6d547e91a1a4f5606

| Data                  | Importance | Source |
|-----------------------|------------|--------|

> | *Track sections*      | Required. I think this one is pretty obvious why. | Extracted from OSM ways with the tag [railway=rail](https://wiki.openstreetmap.org/wiki/Tag:railway%3Drail).​ |

Reviewer 2: ```suggestion
| *Track sections*      | Required. This one's pretty obvious why. | Extracted from OSM ways with the tag [`railway=rail`](https://wiki.openstreetmap.org/wiki/Tag:railway%3Drail).​ |
```

> | *Switches*            | Required. Tracks should be explicitly linked by switches to be connected. | Generated from the angle between the *track sections* that cross each other. This data exist in OSM but we don't use it.​ |

Reviewer 2: ```suggestion
| *Switches*            | Required. Tracks should be explicitly linked by switches to be connected. | Generated from the angle between the *track sections* that cross each other. This data exist in OSM but is not consumed by OSRD.​ |
```

> | *Switches*            | Required. Tracks should be explicitly linked by switches to be connected. | Generated from the angle between the *track sections* that cross each other. This data exist in OSM but is not consumed by OSRD.​ |

Reviewer: ```suggestion
| *Switches*            | Required. Tracks should be explicitly linked by switches to be connected. | Generated from the angle between the *track sections* that cross each other. A specific switch object exist in OSM but is not used because its quality is uneven.​ |
```

Author: Is this why we don't use this data ?

Reviewer: yes and also because our method works

Author: cf895200d38e6de815ad4d7c88b6f40ffa19a78a

> | *Signals*             | Required. They are used in the pathfinding to determine where and how the trains can move on the track network. | Either extracted from OSM nodes with the tag [railway=signal](https://wiki.openstreetmap.org/wiki/Tag:railway%3Dsignal), or generated from the *track sections* data.​ |

Reviewer 2: ```suggestion
| *Signals*             | Required. Signals are used by the pathfinding and the conflict detection to determine where and how the trains can move on the infrastructure.| Either extracted from OSM nodes with the tag [`railway=signal`](https://wiki.openstreetmap.org/wiki/Tag:railway%3Dsignal), either generated by [`osm-to-railjson`](https://osrd.fr/en/docs/content/guides/osm_infra/generation/) (using *track sections* data).​ |
```

> | *Detectors*           | Required. They are used in combinaison with the signal for the pathfinding. | Generated from the *signals* data.​ |

Reviewer 2: ```suggestion
| *Detectors*           | Required. Detectors are used in alongside of the signals for the pathfinding. | Generated from the *signals* data.​ |
```

> | *Operational points*  | Almost required. You can still trace train without it, but most of the application feature will not work. | Extracted from OSM relations with the tag [public_transport=stop_area](https://wiki.openstreetmap.org/wiki/Tag:public_transport%3Dstop_area).​ |

Reviewer 2: ```suggestion
| *Operational points*  | Almost required. Without it, most of the application feature will not work. | Extracted from OSM relations with the tag [`public_transport=stop_area`](https://wiki.openstreetmap.org/wiki/Tag:public_transport%3Dstop_area).​ |
```

> | *Speed sections*      | Nice to have. By default a track section will have a speed limit of 80 km/h, which will impact the realism of the simulation. | Extracted from OSM tag [maxspeed](https://wiki.openstreetmap.org/wiki/Key:maxspeed).​ |

Reviewer 2: ```suggestion
| *Speed sections*      | Nice to have. If not defined, a track section has a speed limit of 80 km/h, which will impact the realism of the simulation. | Extracted from OSM tag [`maxspeed`](https://wiki.openstreetmap.org/wiki/Key:maxspeed).​ |
```

> | *Electrifications*    | Nice to have. Electric rolling stock will not be able to travel on non-electrified tracks. | Extracted from OSM tag [voltage](https://wiki.openstreetmap.org/wiki/Key:voltage).​ |

Reviewer 2: ```suggestion
| *Electrifications*    | Nice to have. Electric rolling stocks are not be able to travel on non-electrified tracks. | Extracted from OSM tag [`voltage`](https://wiki.openstreetmap.org/wiki/Key:voltage).​ |
```

> | *Loading gauges*      | Nice to have. By default the least restrictive one is used, so train can go where they couldn't normaly do. | *Not* extracted, but the data exist [here](https://wiki.openstreetmap.org/wiki/Key:loading_gauge).​ |

Reviewer 2: ```suggestion
| *Loading gauges*      | Nice to have. If not defined, the least restrictive one is used, allowing the trains to run everywhere. | *Not* extracted, but the data exist (see OSM tag [`loading_gauge`](https://wiki.openstreetmap.org/wiki/Key:loading_gauge)).​ |
```

> | *Slopes*              | Nice to have. By default tracks are considered flat, which will impact the realism of the simulation. | This data doesn't exist in OSM. |

Reviewer 2: ```suggestion
| *Slopes*              | Nice to have. If not defined, tracks are considered flat, which will impact the realism of the simulation. | This data doesn't exist in OSM. |
```

> | *Curves*              | Nice to have. By default tracks are considered straight, which will impact the realism of the simulation. | *Not* extracted. The data exist under another format that would be hard to extract. |

Reviewer 2: ```suggestion
| *Curves*              | Nice to have. If not defined, tracks are considered straight, which will impact the realism of the simulation. | *Not* extracted. The data exist under another format that would be hard to extract. |
```

> | *Curves*              | Nice to have. If not defined, tracks are considered straight, which will impact the realism of the simulation. | *Not* extracted. The data exist under another format that would be hard to extract. |

Reviewer: in which tag?

Author: There is no tag. We would have to calculate the curves with the geographical data (the line we use to draw the tracks).

> | *Level crossings*     | Nice to have. It will not affect the simulation or pathfinding, but their closing information can be seen. | *Not* extracted, but the data existe [here](https://wiki.openstreetmap.org/wiki/Tag:railway%3Dlevel_crossing).​ |

Reviewer 2: ```suggestion
| *Level crossings*     | Nice to have. Level crossings do not affect the simulation or pathfinding, but their closing information can be analyzed. | *Not* extracted, but the data exist (see the OSM tag [`level_crossing`](https://wiki.openstreetmap.org/wiki/Tag:railway%3Dlevel_crossing)).​ |
```

> | *Neutral zones*       | Nice to have. | I don't know if the data exist in OSM, but in any case, it isn't extracted. |

Reviewer 2: ```suggestion
| *Neutral zones*       | Nice to have. | We did not explored that yet. |
```

### Solutions

> If *track sections* are missing from your data, there's not much we can do.

Reviewer: ```suggestion
If *track sections* are missing from your data, there's not much we can do. However, this information is usually in very high quality in OSM because it can be traced using aerial imagery (track geometry can be sketchy in tunnels).
```

Author: 6a016b4612d0c2dcad8205d6d547e91a1a4f5606

> But for *signals* we have an option that replace the existing ones with [realisticaly generated *signals*](/en/docs/reference/design-docs/osm-to-railjson#signal-generation).

Reviewer 2: ```suggestion
But for *signals*, a option in the infrastructure generation can be used for replacing the existing ones with [realisticaly generated *signals*](/en/docs/reference/design-docs/osm-to-railjson#signal-generation).
```

> For the *operational points*, their is a few optional fields (*name*, *uic* and *local_track_name*) that we generate to make them easier to use.

Reviewer 2: ```suggestion
For the *operational points*, there are a few optional fields (*name*, *uic* and *local_track_name*) that we generate to make them easier to use.
```

> For the *operational points*, there are a few optional fields (*name*, *uic* and *local_track_name*) that we generate to make them easier to use.

Reviewer: that we generate all the time or only if they are missing?

Author: 6a016b4612d0c2dcad8205d6d547e91a1a4f5606
